pkg/tar/tar_test.go

```diff
@@ -388,33 +410,43 @@ func verifyDirectory(t *testing.T, dir string, files []fileDesc) {
 	if runtime.GOOS == "windows" {
 		for i := range files {
 			files[i].name = filepath.FromSlash(files[i].name)
-			if files[i].mode&0700 == 0400 {
-				files[i].mode &^= 0777
-				files[i].mode |= 0444
-			} else {
-				files[i].mode &^= 0777
+			if files[i].mode&0200 == 0200 {
+				// if the file is user writable make it writable for everyone
 				files[i].mode |= 0666
+			} else {
+				// if the file is only readable, make it readable for everyone
+				// first clear the r/w permission bits
+				files[i].mode &^= 0666
+				// then set r permission for all
+				files[i].mode |= 0444
+			}
+			if files[i].mode.IsDir() {
+				// if the file is a directory, make it executable for everyone
+				files[i].mode |= 0111
```

Comment:
@jim-minter changed this logic to only apply to directories (not files+symlinks) based on the latest failure.